File: 205-isomorphic-strings/isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        if(s.length() != t.length()) return false;
        HashMap <Character, Character> mapst = new HashMap <>();
        HashMap <Character, Character> mapts = new HashMap <>();
        for(int i = 0; i<s.length(); i++){
            char cs = s.charAt(i);
            char ct = t.charAt(i);
            if(mapst.containsKey(cs)){
                if(mapst.get(cs) != ct) return false;
            }
            else{
                mapst.put(cs, ct);
            }
            if(mapts.containsKey (ct)){
                if(mapts.get(ct) != cs) return false;
            }
            else{
                mapts.put(ct, cs);
            }
        }
        return true;
    }

}

File: 205-isomorphic-strings/isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        if(s.size() != t.size()) return false;
        unordered_map <char, char> st;
        unordered_map <char, char> ts;
        for(int i = 0; i <s.size(); i++){
            char cs = s[i];
            char ct = t[i];
            // if(st.find(cs) == st.end()) st[cs] = ct;
            // else if(st[cs] != ct) return false;

            // if(ts.find(ct) == ts.end()) ts[ct] = cs;
            // else if(ts[ct] != cs) return false;

            if(st.count(cs) && st[cs] != ct) return false;
            if(ts.count(ct) && ts[ct] != cs) return false;

            st[cs] = ct;
            ts[ct] = cs;
        }
        return true;
    }
};